spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/query/Query.java

```diff
@@ -517,13 +518,13 @@ public boolean isSorted() {
 			}
 		};
 
-		target.criteria.putAll(source.criteria);
-		target.skip = source.skip;
-		target.limit = source.limit;
-		target.sort = Sort.unsorted().and(source.sort);
-		target.hint = source.hint;
-		target.collation = source.collation;
-		target.restrictedTypes.addAll(source.restrictedTypes);
+		Query theQuery = ProxyUtils.unwrapTargetSource(source);
+
+		target.skip = theQuery.skip;
```

Comment:
How about using accessor methods instead of field access? That way, we can entirely omit any kind of proxy handling.

Reply:
good point :)